coordinator/coordinator.go

```diff
@@ -360,63 +502,89 @@ func (c *Coordinator) runFeatureMaterializeJob(resID metadata.ResourceID) error
 	if err != nil {
 		return fmt.Errorf("could not fetch online provider: %w", err)
 	}
-	p, err := provider.Get(provider.Type(sourceProvider.Type()), sourceProvider.SerializedConfig())
-	if err != nil {
-		return fmt.Errorf("could not get offline provider config: %w", err)
-	}
-	sourceStore, err := p.AsOfflineStore()
-	if err != nil {
-		return fmt.Errorf("could not use store as offline store: %w", err)
-	}
+	// p, err := provider.Get(provider.Type(sourceProvider.Type()), sourceProvider.SerializedConfig())
+	// if err != nil {
+	// 	return err
+	// }
+	//sourceStore, err := p.AsOfflineStore()
+	// if err != nil {
+	// 	return err
+	// }
 	featureProvider, err := feature.FetchProvider(c.Metadata, context.Background())
 	if err != nil {
 		return fmt.Errorf("could not fetch  onlineprovider: %w", err)
 	}
-	p, err = provider.Get(provider.Type(featureProvider.Type()), featureProvider.SerializedConfig())
+	// p, err = provider.Get(provider.Type(featureProvider.Type()), featureProvider.SerializedConfig())
+	// if err != nil {
+	// 	return err
+	// }
+	//featureStore, err := p.AsOnlineStore()
+	// if err != nil {
+	// 	return err
+	// }
+
+	type JobCloud string
+
+	const (
+		KubernetesMaterializeRunner JobCloud = "KUBERNETES"
+		LocalMaterializeRunner      JobCloud = "LOCAL"
+	)
+	materializedRunnerConfig := runner.MaterializedRunnerConfig{
+		OnlineType:    provider.Type(featureProvider.Type()),
+		OfflineType:   provider.Type(sourceProvider.Type()),
+		OnlineConfig:  featureProvider.SerializedConfig(),
+		OfflineConfig: sourceProvider.SerializedConfig(),
+		ResourceID:    provider.ResourceID{Name: resID.Name, Variant: resID.Variant, Type: provider.Feature},
+		VType:         provider.ValueType(featureType),
+		Cloud:         runner.LocalMaterializeRunner,
+		IsUpdate:      false,
+	}
+	serialized, err := materializedRunnerConfig.Serialize()
 	if err != nil {
 		return fmt.Errorf("could not get online provider config: %w", err)
 	}
-	featureStore, err := p.AsOnlineStore()
+	jobRunner, err := c.Spawner.GetJobRunner(runner.MATERIALIZE, serialized, c.EtcdClient.Endpoints(), resID)
 	if err != nil {
 		return fmt.Errorf("could not use store as online store: %w", err)
 	}
-	srcID := provider.ResourceID{
-		Name:    sourceNameVariant.Name,
-		Variant: sourceNameVariant.Variant,
-	}
-	srcName, err := provider.GetTransformationName(srcID)
-	if err != nil {
-		return fmt.Errorf("transform name err: %w", err)
-	}
-
-	featID := provider.ResourceID{
-		Name:    resID.Name,
-		Variant: resID.Variant,
-		Type:    provider.Feature,
-	}
-	tmpSchema := feature.LocationColumns().(metadata.ResourceVariantColumns)
-	schema := provider.ResourceSchema{
-		Entity:      tmpSchema.Entity,
-		Value:       tmpSchema.Value,
-		TS:          tmpSchema.TS,
-		SourceTable: srcName,
-	}
-	c.Logger.Debugw("Creating Resource Table", "id", featID, "schema", schema)
-	_, err = sourceStore.RegisterResourceFromSourceTable(featID, schema)
```

Comment:
Are features being registered somewhere else?

Reply:
they aren't... It turns out the test I wrote for this assumed they were already registered in the original table for some reason. Guess I didn't think properly that upon materialization the feature would not exist at all and need to be identified column-wise in the table, then materialized, then copied. I will uncomment these.